File: Data Structures & Algorithms/max-consecutive-ones/submission-0.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max = 0;
        int count = 0;
        for (auto i: nums)
        {
            if (i)
            {
                count++;
                if (count > max)
                    max = count;
            }
            else
                count = 0;
        }
        return max;
    }
};

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-0.ts
class Solution {
    /**
     * @param {number[]} arr
     * @return {number[]}
     */
    findMax(i: number, arr: number[]): number
    {
        let max = 0;
        for (; i < arr.length; i++)
        {
            if (arr[i] > max)
                max = arr[i];
        }
        return (max);
    }

    replaceElements(arr: number[]): number[] 
    {
        let max = this.findMax(1, arr);
        let i = 0;
        while (i < arr.length - 1)
        {
            if (arr[i] === max)
                max = this.findMax(i + 1, arr);
            arr[i] = max;
            i++;
        }
        arr[i] = -1;
        return arr;
    }
}
